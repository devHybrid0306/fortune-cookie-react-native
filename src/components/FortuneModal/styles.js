import { StyleSheet } from 'react-native';
import { RatioW } from '../../utils';

const styles = StyleSheet.create({
  modal: {
    margin: 0,
  },
  container: {
    flex: 1,
    padding: 20 * RatioW,
    alignItems: 'center',
    justifyContent: 'center',
  },
  text: {
    fontSize: 25 * RatioW,
    color: '#ffffff',
    textAlign: 'center',
  },
  dateContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'rgba(255, 255, 255, .2)',
    width: 120 * RatioW,
    borderRadius: 10 * RatioW,
    position: 'absolute',
    bottom: 50 * RatioW,
  },
  date: {
    fontSize: 12 * RatioW,
    color: '#ffffff',
    textAlign: 'center',
    padding: 3 * RatioW,
  },
});

export default styles;
