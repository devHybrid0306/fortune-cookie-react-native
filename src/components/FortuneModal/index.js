import React from 'react';
import { Text, View } from 'react-native';
import Modal from 'react-native-modal';

import { getFormattedDate, RatioW } from '../../utils';
import CircleButton from '../CircleButton';
import styles from './styles';

const FortuneModal = ({ isVisible, closeModal, data }) => {
  return (
    <Modal
      isVisible={isVisible}
      onBackButtonPress={closeModal}
      customBackdrop={<View style={{ flex: 1, backgroundColor: data?.mColor }} />}
      backdropOpacity={1}
      animationIn={'zoomIn'}
      animationOut={'zoomOut'}
      style={styles.modal}>
      <View style={{ ...styles.container, backgroundColor: data?.mColor }}>
        <View style={{ position: 'absolute', top: 50 * RatioW, right: 20 * RatioW }}>
          <CircleButton
            handleButton={closeModal}
            size={40}
            bkgColor={'#ffffff'}
            text={'âœ•'}
            textColor={data?.mColor}
            textSize={25}
          />
        </View>
        <Text style={styles.text}>{data?.text}</Text>
        <View style={styles.dateContainer}>
          <Text style={styles.date}>{getFormattedDate(data?.date)}</Text>
        </View>
      </View>
    </Modal>
  );
};

export default FortuneModal;
