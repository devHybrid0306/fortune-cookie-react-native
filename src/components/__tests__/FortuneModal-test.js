import React from 'react';
import renderer from 'react-test-renderer';
import FortuneModal from '../FortuneModal';

test('renders correctly', () => {
  const tree = renderer.create(<FortuneModal />).toJSON();
  expect(tree).toMatchSnapshot();
});
