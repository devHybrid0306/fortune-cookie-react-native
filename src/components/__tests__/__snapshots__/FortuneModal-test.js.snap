// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  animationType="none"
  deviceHeight={null}
  deviceWidth={null}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  panResponderThreshold={4}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  statusBarTranslucent={false}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={false}
>
  <View
    forwardedRef={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": undefined,
          "flex": 1,
        }
      }
    />
  </View>
  <View
    deviceHeight={null}
    deviceWidth={null}
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    panResponderThreshold={4}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 0,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": undefined,
          "flex": 1,
          "justifyContent": "center",
          "padding": 40,
        }
      }
    >
      <View
        style={
          Object {
            "position": "absolute",
            "right": 40,
            "top": 100,
          }
        }
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderRadius": 40,
              "elevation": 14,
              "height": 80,
              "justifyContent": "center",
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 7,
                "width": 0,
              },
              "shadowOpacity": 0.41,
              "shadowRadius": 9.11,
              "width": 80,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 25,
              }
            }
          >
            âœ•
          </Text>
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 50,
            "textAlign": "center",
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(255, 255, 255, .2)",
            "borderRadius": 20,
            "bottom": 100,
            "justifyContent": "center",
            "position": "absolute",
            "width": 240,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 24,
              "padding": 6,
              "textAlign": "center",
            }
          }
        >
          July 7 2021
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
